### PR TITLE
experiment env uses corev1.EnvVar instead of the type of name/value pairs

### PR DESCRIPTION
**Pull request by kazukousen**
Previously, experiment env can only use plain text.
This patch allows to use EnvVar at CR schema so that it can get value from secret/configmap/downwardAPI

https://github.com/litmuschaos/litmus/issues/2368 depends on this PR.

Signed-off-by: nitta-kazuki <mmchari.0228@gmail.com>

**Checklist:**
-   [ ] Fixes #
-   [ ] Labelled this PR & related issue with `documentation` tag
-   [ ] PR messages has document related information
-   [ ] Labelled this PR & related issue with `breaking-changes` tag
-   [ ] PR messages has breaking changes related information
-   [ ] Labelled this PR & related issue with `requires-upgrade` tag
-   [ ] PR messages has upgrade related information
-   [ ] Commit has unit tests
-   [ ] Commit has integration tests

### CODE REVIEW

**Comment by ksatchit** [comment]
@kazukousen thanks for creating this PR. Moving the ENVs to corev1.EnvVar over a custom struct is a good idea. The CRD validations placed for it seem comprehensive too. Could you please share the results of some manually executed tests for - usage of valueFrom & prevention of engine creation when both values/valueFrom is provided?

**Comment by ksatchit** [comment]
> /run-e2e-all
> 
> **Test Status:** The e2e test has been started please wait for the results ...
> **Test Failed:** Some tests are failed please check
> **Run ID:** [399955248](https://github.com/litmuschaos/chaos-operator/actions/runs/399955248)

@uditgaurav we may need to fix the set:env steps here!

File: pkg/apis/litmuschaos/v1alpha1/chaosexperiment_types.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ type ExperimentDef struct {
 	// List of Permission needed for a service account to execute experiment
 	Permissions []rbacV1.PolicyRule `json:"permissions"`
 	// List of ENV vars passed to executor pod
```
Comment (ksatchit) [inline pkg/apis/litmuschaos/v1alpha1/chaosexperiment_types.go:75]: Makes sense to call this chaosexperiment pod

File: deploy/crds/chaosexperiment_crd.yaml [hunk 2]
```diff
@@ -43,13 +43,112 @@ spec:
                     type: string
                 env:
                   type: array
+                  description: List of ENV vars passed to executor pod
```
Comment (ksatchit) [inline deploy/crds/chaosexperiment_crd.yaml:46]: Ref: https://github.com/litmuschaos/chaos-operator/pull/309/files#r535906743

File: pkg/apis/litmuschaos/v1alpha1/chaosexperiment_types.go (hunk 1 above)
Comment (kazukousen) [reply pkg/apis/litmuschaos/v1alpha1/chaosexperiment_types.go:75]: you mean it's this? https://github.com/litmuschaos/chaos-operator/pull/309/commits/659798a2ebfde7e2fa09369188640790f2ec5804

**Comment by ksatchit** [comment]
@kazukousen seeing this error upon attempting to apply the CRDs from chaos_crds.yaml - looks like a missed param: 

kubectl apply -f https://raw.githubusercontent.com/litmuschaos/chaos-operator/659798a2ebfde7e2fa09369188640790f2ec5804/deploy/chaos_crds.yaml

```
Error from server (Invalid): error when creating "https://raw.githubusercontent.com/litmuschaos/chaos-operator/659798a2ebfde7e2fa09369188640790f2ec5804/deploy/chaos_crds.yaml": CustomResourceDefinition.apiextensions.k8s.io "chaosengines.litmuschaos.io" is invalid: spec.validation.openAPIV3Schema.type: Required value: must not be empty at the root
Error from server (Invalid): error when creating "https://raw.githubusercontent.com/litmuschaos/chaos-operator/659798a2ebfde7e2fa09369188640790f2ec5804/deploy/chaos_crds.yaml": CustomResourceDefinition.apiextensions.k8s.io "chaosexperiments.litmuschaos.io" is invalid: spec.validation.openAPIV3Schema.type: Required value: must not be empty at the root
```

**Comment by kazukousen** [comment]
@ksatchit 
sorry, i fixed it.
remove `x-kubernetes-int-or-string: true` and the validation will pass https://github.com/litmuschaos/chaos-operator/pull/309/commits/bcf9b978423f7d515e78e4ec9c143db5b95de4a6

thank you for your kind review.
> Could you please share the results of some manually executed tests

i am working on it.

**Comment by kazukousen** [comment]
To achieve this feature, a change in litmus-runner is required.
https://github.com/litmuschaos/chaos-runner/blob/master/pkg/utils/builders.go#L85
I am confused how to do the e2e tests manually.

**Comment by kazukousen** [comment]
the EnvVar `description` field in crds.yaml seems redundant. should it be removed?

**Comment by ksatchit** [comment]
> To achieve this feature, a change in litmus-runner is required.
> https://github.com/litmuschaos/chaos-runner/blob/master/pkg/utils/builders.go#L85
> I am confused how to do the e2e tests manually.

Hey @kazukousen ! By manual test was referring to the basic successful install of the CRD spec and a potential validation against providing both value and valueFrom (https://github.com/litmuschaos/chaos-operator/pull/309#issuecomment-738563269). And a general regression test re: experiment execution. As regards use of valueFrom, we certainly do need chaos-runner changes. 

Coming to the results based on https://github.com/litmuschaos/chaos-operator/commit/bcf9b978423f7d515e78e4ec9c143db5b95de4a6: 

- I am able to successfully install the CRDs & run experiments
- wrt making `value` & `valueFrom` mutually exclusive seems like it is not supported by openAPI validation yet: https://github.com/OAI/OpenAPI-Specification/issues/256. So we can document accordingly. 

Re: description it should be ok @kazukousen

**Comment by ksatchit** [comment]
Merging the PR! Thank you once again @kazukousen looking forward to runner changes.

**Comment by kazukousen** [comment]
thank you again for the your kind review! @ksatchit 

i am working on it, but i decided to wait for the refactoring work here.
https://github.com/litmuschaos/chaos-runner/pull/117

**Pull request merged**